asset-transfer-private-data/chaincode-go/private_asset_transfer.go

```diff
@@ -0,0 +1,497 @@
+type AssetPrivateDetails struct {
+	ID             string `json:"asset_id"`
+	AppraisedValue int    `json:"appraised_value"`
+}
+
+const assetCollection = "assetCollection"
+
+type SmartContract struct {
+	contractapi.Contract
+}
+
+// CreateAsset creates a new asset by placing the main asset details in the assetCollection
+// that can be read by both organizations. The appraisal value is stored in the owners org specific collection.
+func (s *SmartContract) CreateAsset(ctx contractapi.TransactionContextInterface) error {
+
+	// Get new asset from transient map
+	transMap, err := ctx.GetStub().GetTransient()
+	if err != nil {
+		return fmt.Errorf("Error getting transient: " + err.Error())
+	}
+
+	// Asset properties are private, therefore they get passed in transient field
+	transientAssetJSON, ok := transMap["asset_properties"]
+	if !ok {
+		return fmt.Errorf("asset not found in the transient map")
+	}
+
+	type assetTransientInput struct {
+		Type 		 				string `json:"type"`  //Type is used to distinguish the various types of objects in state database
+		ID             	string `json:"asset_id"`
+		Color          	string `json:"color"`
+		Size           	int    `json:"size"`
+		AppraisedValue 	int    `json:"appraised_value"`
+	}
+
+	var assetInput assetTransientInput
+	err = json.Unmarshal(transientAssetJSON, &assetInput)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal JSON: %s", err.Error())
+	}
+
+	if len(assetInput.Type) == 0 {
+		return fmt.Errorf("type field must be a non-empty string")
+	}
+	if len(assetInput.ID) == 0 {
+		return fmt.Errorf("asset_id field must be a non-empty string")
+	}
+	if len(assetInput.Color) == 0 {
+		return fmt.Errorf("color field must be a non-empty string")
+	}
+	if assetInput.Size <= 0 {
+		return fmt.Errorf("size field must be a positive integer")
+	}
+	if assetInput.AppraisedValue <= 0 {
+		return fmt.Errorf("AppraisedValue field must be a positive integer")
+	}
+
+	// Check if asset already exists
+	assetAsBytes, err := ctx.GetStub().GetPrivateData(assetCollection, assetInput.ID)
+	if err != nil {
+		return fmt.Errorf("Failed to get asset: " + err.Error())
+	} else if assetAsBytes != nil {
+		fmt.Println("This asset already exists: " + assetInput.ID)
+		return fmt.Errorf("This asset already exists: " + assetInput.ID)
+	}
+
+	// Get ID of submitting client identity
+	clientID, err := ctx.GetClientIdentity().GetID()
+	if err != nil {
+		return fmt.Errorf("failed to get verified OrgID: %s", err.Error())
+	}
+
+	// Make submitting client the owner
+	asset := &Asset{
+		Type: 			assetInput.Type,
+		ID:       	assetInput.ID,
+		Color:      assetInput.Color,
+		Size:       assetInput.Size,
+		Owner:      clientID,
+	}
+	assetJSONasBytes, err := json.Marshal(asset)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	// Save asset to private data collection
+	err = ctx.GetStub().PutPrivateData(assetCollection, assetInput.ID, assetJSONasBytes)
```

Comment:
assetCollection is not declared

It will be easier for comprehension if all collection names are declared on top as constants, and used in both go files. So up in the top
```
const collectionNameSharedBetweenOrgs := "assetCollection"  
const collectionNameOrg1Private := "Org1MSPPrivateCollection"  
const collectionNameOrg2Private := "Org2MSPPrivateCollection"  

```

```suggestion
	err = ctx.GetStub().PutPrivateData(collectionNameSharedBetweenOrgs, assetInput.ID, assetJSONasBytes)
```

Reply:
Actually asset collection is declared at the top: `const assetCollection = "assetCollection"`

The Org1 or Org2 private collection is retrieved by the smart contract depending on context.